File: Assignment/2016-2017/C#/Assignment - PART 1/AssignmentComplete/ITruck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace AssignmentComplete
{
  public interface ITruck : IComponent
  {
    IContainer Container { get; }
    Vector2 Position { get; }
    Vector2 Velocity { get; }
    void StartEngine();
    void AddContainer(IContainer container);
  }
}

File: Assignment/2016-2017/Java/Assignment-PART1/src/truck/ITruck.java
package truck;

import component.IComponent;
import container.IContainer;
import javafx.geometry.Point2D;

public interface ITruck extends IComponent {
    void startEngine();

    void addContainer(IContainer container);

    IContainer getContainer();

    Point2D getPosition();

    Point2D getVelocity();
}
